src/nvim/extmark.c

```diff
@@ -132,22 +132,14 @@ void extmark_set(buf_T *buf, uint32_t ns_id, uint32_t *idp, int row, colnr_T col
   } else if (decor) {
     mark.hl_id = decor->hl_id;
     // workaround: see above
-    mark.flags = (uint16_t)(mark.flags | (decor->hl_eol ? (uint16_t)MT_FLAG_HL_EOL : (uint16_t)0));
+    mark.flags = (uint16_t)(mark.flags | (decor->hl_eol ? (uint16_t)MT_FLAG_HL_EOL : (uint16_t)0)
+                            | (no_undo ? (uint16_t)MT_FLAG_NO_UNDO : 0));
```

Comment:
Perhaps invert the flag/boolean names as well?
I also think we need to adjust all paths that set `(old_)mark.flags` in `extmark_set()`, e.g. add an `undo` argument to `mt_flags()`. Otherwise this doesn't work when revising marks/for marks with a decor right?

```vim
" test.vim
call setline(1, repeat(['aaa bbb ccc'], 5))
let ns = nvim_create_namespace('')
call nvim_buf_set_extmark(0, ns, 2, 4, { 'id':2, 'undo_restore':0, 'sign_text':'S1', 'hl_group' : 'Error', 'end_col':7, 'end_row':3 })
```